src/marlin_stubs/sdcard/M20-M30_M32-M34.cpp

```diff
@@ -119,7 +120,17 @@ void GcodeSuite::M29() {
 
 // M30 - Delete a file on the SD card
 void GcodeSuite::M30() {
-    // TODO
+    char filepath[FILE_PATH_MAX_LEN] = "/usb/";
+    strlcat(filepath, parser.string_arg, FILE_PATH_MAX_LEN);
+    DeleteResult result = remove_file(filepath);
+    if (result == DeleteResult::Success) {
+        SERIAL_ECHOPGM("File deleted:");
+        SERIAL_ECHOLN(parser.string_arg);
+    } else {
+        SERIAL_ECHOPGM("Deletion failed, File: ");
+        SERIAL_ECHO(parser.string_arg);
+        SERIAL_ECHOLN(".");
```

Comment:
Thy is there "." at the end of one response and on the other one? If this was unified, you could put `SERIAL_ECHOLN(parser.string_arg);` outside of the branch and save some flash.

You could also possibly save some flash by doing `SERIAL_ECHOPGM(result == success ? "File deleted:": "Deletion failed:");`

Reply:
I was doing a cnp from the Prusa-Firmware, but you're right. Changing the code accordingly :)

Reply:
Just a note, I guess the octoprint will be very picky about string format, so we should try mimicking Marlin as much as possible.

Reply:
Yeah, I will check the octoprint code and make sure string format matches.

Reply:
Doesn't look like octoprint does check the response, instead it issues a M20 code after sending M30, see [here](https://github.com/OctoPrint/OctoPrint/blob/d79a0d20f3f1c7f2edb56dedda3b70267a937e65/src/octoprint/util/comm.py#L2041), so I think we should be fine.